blaze/interactive.py

```diff
@@ -281,9 +281,24 @@ def into(a, b, **kwargs):
 def table_length(expr):
     try:
         return expr._len()
-    except TypeError:
+    except ValueError:
         return compute(expr.count())
 
 
 Expr.__repr__ = expr_repr
 Expr.__len__ = table_length
+
+def intonumpy(data, dtype=None, **kwargs):
+    # TODO: Don't ignore other kwargs like copy
+    result = into(np.ndarray, data)
+    if dtype:
+        result = result.astype(dtype)
+    return result
+
+Expr.__array__ = intonumpy
+Expr.__int__ = lambda x: int(compute(x))
```

Comment:
i think these could also be written as `into(int)`, etc. not a big deal, just a comment that can probably be ignored